datafusion/expr/src/function.rs

```diff
@@ -222,6 +222,14 @@ pub fn return_type(
             _ => Ok(DataType::Float64),
         },
 
+        BuiltinScalarFunction::Struct => {
+            let fields = input_expr_types
+                .iter()
+                .map(|x| Field::new("item", x.clone(), true))
```

Comment:
This looks like it is generating a struct where all of the fields have the same name. I would expect that to cause an error.

Reply:
Fix, It should only need the DataType info, no need for fields name